### ISSUE TITLE
[Feature request] Multi-class classification objective function using OneVersusOne method

### ISSUE THREAD

**Issue by ChrisZheng1985** [dmlc/xgboost#6242]
Hi,

For multi-class classification, we have the objective functions of 'multi:softmax' and 'multi"softprob', which basically are using the OneVersusOthers method (correct me if wrong).

In a lot of modeling execrise, for instance, we need to classify A, B, C1 and C2 where C1 and C2 are classes that share a lot similar characteristics. Thus when we have our primary focus on predicting C1, we use the OneVersusOther method, the model is trying to predict C1 versus (A, B and C2) and pick up some variables that may not be as reasonable as the one using the OneVersusOne like C1 vs A.

To make the request clear, it would be great if we have an additional objective function of 'multi:OneVursusOne' , that:
1. we can specify the reference group;
2. the optimization is using 1 versus 1 method. for example in the above example, it basically will evaluate C1 vs A, C2 vs A and B vs A.

best,
Chris

**Comment by trivialfis** [comment]
It's unclear to me what exactly do you want XGBoost to do.  Could you please elaborate on:

- The mathematical definition of reference group.
- For point 2, this seems to be combinatorial explosion?

**Comment by trivialfis** [comment]
That make sense.  But still, what's reference group?

**Comment by ChrisZheng1985** [comment]
@trivialfis 
Sorry for the confusion,

For example, when we have classes of A, B, C, D. Suppose use chooses 'The reference group = D', then the objective function will evaluate a series of 'binomial' classification of A vs D, B vs D, and C vs D. And then convert each of the logodd to the probability of the four classes. 

A more details can also refer to the below:
section 'As a set of independent binary regressions' in https://en.wikipedia.org/wiki/Multinomial_logistic_regression

Let me know if this is not very clear.

best,
Chris
